<?xml version="1.0" encoding="UTF-8"?>
<cvrfdoc xmlns="http://www.icasi.org/CVRF/schema/cvrf/1.1" xmlns:cvrf="http://www.icasi.org/CVRF/schema/cvrf/1.1">
  <DocumentTitle xml:lang="en">Security update for python-Django</DocumentTitle>
  <DocumentType>SUSE Patch</DocumentType>
  <DocumentPublisher Type="Vendor">
    <ContactDetails>security@suse.de</ContactDetails>
    <IssuingAuthority>SUSE Security Team</IssuingAuthority>
  </DocumentPublisher>
  <DocumentTracking>
    <Identification>
      <ID>openSUSE-SU-2015:0643-1</ID>
    </Identification>
    <Status>Final</Status>
    <Version>1</Version>
    <RevisionHistory>
      <Revision>
        <Number>1</Number>
        <Date>2015-03-25T13:54:57Z</Date>
        <Description>current</Description>
      </Revision>
    </RevisionHistory>
    <InitialReleaseDate>2015-03-25T13:54:57Z</InitialReleaseDate>
    <CurrentReleaseDate>2015-03-25T13:54:57Z</CurrentReleaseDate>
    <Generator>
      <Engine>cve-database/bin/generate-cvrf.pl</Engine>
      <Date>2017-02-24T01:00:00Z</Date>
    </Generator>
  </DocumentTracking>
  <DocumentNotes>
    <Note Title="Topic" Type="Summary" Ordinal="1" xml:lang="en">Security update for python-Django</Note>
    <Note Title="Details" Type="General" Ordinal="2" xml:lang="en">python-django was updated to 1.6.11 to fix security issues and non-security bugs.

THe following vulnerabilities were fixed:

* Made is_safe_url() reject URLs that start with control characters to mitigate possible XSS attack via user-supplied redirect URLs (bnc#923176, CVE-2015-2317)
* Fixed an infinite loop possibility in strip_tags() (bnc#923172, CVE-2015-2316)
* WSGI header spoofing via underscore/dash conflation (bnc#913053, CVE-2015-0219)
* Mitigated possible XSS attack via user-supplied redirect URLs
* Denial-of-service attack against ``django.views.static.serve``   (bnc#913056, CVE-2015-0221)
* Database denial-of-service with ``ModelMultipleChoiceField`` (bnc#913055, CVE-2015-0222)

The update also contains fixes for non-security bugs, functional and stability issues.
</Note>
    <Note Title="Terms of Use" Type="Legal Disclaimer" Ordinal="3" xml:lang="en">The CVRF data is provided by SUSE under the Creative Commons License 4.0 with Attribution for Non-Commercial usage (CC-BY-NC-4.0).</Note>
  </DocumentNotes>
  <DocumentDistribution xml:lang="en">Copyright SUSE LLC under the Creative Commons License 4.0 with Attribution for Non-Commercial usage (CC-BY-NC-4.0)</DocumentDistribution>
  <DocumentReferences>
    <Reference Type="Self">
      <URL>http://lists.opensuse.org/opensuse-updates/2015-04/msg00001.html</URL>
      <Description>E-Mail link for openSUSE-SU-2015:0643-1</Description>
    </Reference>
    <Reference Type="Self">
      <URL>https://www.suse.com/support/security/rating/</URL>
      <Description>SUSE Security Ratings</Description>
    </Reference>
  </DocumentReferences>
  <ProductTree xmlns="http://www.icasi.org/CVRF/schema/prod/1.1">
    <Branch Type="Product Family" Name="openSUSE 13.2">
      <Branch Type="Product Name" Name="openSUSE 13.2">
        <FullProductName ProductID="openSUSE 13.2">openSUSE 13.2</FullProductName>
      </Branch>
    </Branch>
    <Branch Type="Product Version" Name="python-Django-1.6.11-3.4.1">
      <FullProductName ProductID="python-Django-1.6.11-3.4.1">python-Django-1.6.11-3.4.1</FullProductName>
    </Branch>
    <Relationship ProductReference="python-Django-1.6.11-3.4.1" RelationType="Default Component Of" RelatesToProductReference="openSUSE 13.2">
      <FullProductName ProductID="openSUSE 13.2:python-Django-1.6.11-3.4.1">python-Django-1.6.11-3.4.1 as a component of openSUSE 13.2</FullProductName>
    </Relationship>
  </ProductTree>
  <Vulnerability xmlns="http://www.icasi.org/CVRF/schema/vuln/1.1" Ordinal="1">
    <Notes>
      <Note Title="Vulnerability Description" Type="General" Ordinal="1" xml:lang="en">Django before 1.4.18, 1.6.x before 1.6.10, and 1.7.x before 1.7.3 allows remote attackers to spoof WSGI headers by using an _ (underscore) character instead of a - (dash) character in an HTTP header, as demonstrated by an X-Auth_User header.</Note>
    </Notes>
    <CVE>CVE-2015-0219</CVE>
    <ProductStatuses>
      <Status Type="Fixed">
        <ProductID>openSUSE 13.2:python-Django-1.6.11-3.4.1</ProductID>
      </Status>
    </ProductStatuses>
    <Threats>
      <Threat Type="Impact">
        <Description>moderate</Description>
      </Threat>
    </Threats>
    <Remediations>
      <Remediation Type="Vendor Fix">
        <Description xml:lang="en">Please Install the update.</Description>
        <URL>http://lists.opensuse.org/opensuse-updates/2015-04/msg00001.html</URL>
      </Remediation>
    </Remediations>
    <References>
      <Reference>
        <URL>https://www.suse.com/security/cve/CVE-2015-0219.html</URL>
        <Description>CVE-2015-0219</Description>
      </Reference>
      <Reference>
        <URL>https://bugzilla.suse.com/913053</URL>
        <Description>SUSE Bug 913053</Description>
      </Reference>
      <Reference>
        <URL>https://bugzilla.suse.com/919763</URL>
        <Description>SUSE Bug 919763</Description>
      </Reference>
    </References>
  </Vulnerability>
  <Vulnerability xmlns="http://www.icasi.org/CVRF/schema/vuln/1.1" Ordinal="2">
    <Notes>
      <Note Title="Vulnerability Description" Type="General" Ordinal="1" xml:lang="en">The django.views.static.serve view in Django before 1.4.18, 1.6.x before 1.6.10, and 1.7.x before 1.7.3 reads files an entire line at a time, which allows remote attackers to cause a denial of service (memory consumption) via a long line in a file.</Note>
    </Notes>
    <CVE>CVE-2015-0221</CVE>
    <ProductStatuses>
      <Status Type="Fixed">
        <ProductID>openSUSE 13.2:python-Django-1.6.11-3.4.1</ProductID>
      </Status>
    </ProductStatuses>
    <Threats>
      <Threat Type="Impact">
        <Description>moderate</Description>
      </Threat>
    </Threats>
    <Remediations>
      <Remediation Type="Vendor Fix">
        <Description xml:lang="en">Please Install the update.</Description>
        <URL>http://lists.opensuse.org/opensuse-updates/2015-04/msg00001.html</URL>
      </Remediation>
    </Remediations>
    <References>
      <Reference>
        <URL>https://www.suse.com/security/cve/CVE-2015-0221.html</URL>
        <Description>CVE-2015-0221</Description>
      </Reference>
      <Reference>
        <URL>https://bugzilla.suse.com/913056</URL>
        <Description>SUSE Bug 913056</Description>
      </Reference>
      <Reference>
        <URL>https://bugzilla.suse.com/919763</URL>
        <Description>SUSE Bug 919763</Description>
      </Reference>
    </References>
  </Vulnerability>
  <Vulnerability xmlns="http://www.icasi.org/CVRF/schema/vuln/1.1" Ordinal="3">
    <Notes>
      <Note Title="Vulnerability Description" Type="General" Ordinal="1" xml:lang="en">ModelMultipleChoiceField in Django 1.6.x before 1.6.10 and 1.7.x before 1.7.3, when show_hidden_initial is set to True, allows remote attackers to cause a denial of service by submitting duplicate values, which triggers a large number of SQL queries.</Note>
    </Notes>
    <CVE>CVE-2015-0222</CVE>
    <ProductStatuses>
      <Status Type="Fixed">
        <ProductID>openSUSE 13.2:python-Django-1.6.11-3.4.1</ProductID>
      </Status>
    </ProductStatuses>
    <Threats>
      <Threat Type="Impact">
        <Description>moderate</Description>
      </Threat>
    </Threats>
    <Remediations>
      <Remediation Type="Vendor Fix">
        <Description xml:lang="en">Please Install the update.</Description>
        <URL>http://lists.opensuse.org/opensuse-updates/2015-04/msg00001.html</URL>
      </Remediation>
    </Remediations>
    <References>
      <Reference>
        <URL>https://www.suse.com/security/cve/CVE-2015-0222.html</URL>
        <Description>CVE-2015-0222</Description>
      </Reference>
      <Reference>
        <URL>https://bugzilla.suse.com/913055</URL>
        <Description>SUSE Bug 913055</Description>
      </Reference>
      <Reference>
        <URL>https://bugzilla.suse.com/919763</URL>
        <Description>SUSE Bug 919763</Description>
      </Reference>
    </References>
  </Vulnerability>
  <Vulnerability xmlns="http://www.icasi.org/CVRF/schema/vuln/1.1" Ordinal="4">
    <Notes>
      <Note Title="Vulnerability Description" Type="General" Ordinal="1" xml:lang="en">The utils.html.strip_tags function in Django 1.6.x before 1.6.11, 1.7.x before 1.7.7, and 1.8.x before 1.8c1, when using certain versions of Python, allows remote attackers to cause a denial of service (infinite loop) by increasing the length of the input string.</Note>
    </Notes>
    <CVE>CVE-2015-2316</CVE>
    <ProductStatuses>
      <Status Type="Fixed">
        <ProductID>openSUSE 13.2:python-Django-1.6.11-3.4.1</ProductID>
      </Status>
    </ProductStatuses>
    <Threats>
      <Threat Type="Impact">
        <Description>moderate</Description>
      </Threat>
    </Threats>
    <Remediations>
      <Remediation Type="Vendor Fix">
        <Description xml:lang="en">Please Install the update.</Description>
        <URL>http://lists.opensuse.org/opensuse-updates/2015-04/msg00001.html</URL>
      </Remediation>
    </Remediations>
    <References>
      <Reference>
        <URL>https://www.suse.com/security/cve/CVE-2015-2316.html</URL>
        <Description>CVE-2015-2316</Description>
      </Reference>
      <Reference>
        <URL>https://bugzilla.suse.com/923172</URL>
        <Description>SUSE Bug 923172</Description>
      </Reference>
    </References>
  </Vulnerability>
  <Vulnerability xmlns="http://www.icasi.org/CVRF/schema/vuln/1.1" Ordinal="5">
    <Notes>
      <Note Title="Vulnerability Description" Type="General" Ordinal="1" xml:lang="en">The utils.http.is_safe_url function in Django before 1.4.20, 1.5.x, 1.6.x before 1.6.11, 1.7.x before 1.7.7, and 1.8.x before 1.8c1 does not properly validate URLs, which allows remote attackers to conduct cross-site scripting (XSS) attacks via a control character in a URL, as demonstrated by a \x08javascript: URL.</Note>
    </Notes>
    <CVE>CVE-2015-2317</CVE>
    <ProductStatuses>
      <Status Type="Fixed">
        <ProductID>openSUSE 13.2:python-Django-1.6.11-3.4.1</ProductID>
      </Status>
    </ProductStatuses>
    <Threats>
      <Threat Type="Impact">
        <Description>moderate</Description>
      </Threat>
    </Threats>
    <Remediations>
      <Remediation Type="Vendor Fix">
        <Description xml:lang="en">Please Install the update.</Description>
        <URL>http://lists.opensuse.org/opensuse-updates/2015-04/msg00001.html</URL>
      </Remediation>
    </Remediations>
    <References>
      <Reference>
        <URL>https://www.suse.com/security/cve/CVE-2015-2317.html</URL>
        <Description>CVE-2015-2317</Description>
      </Reference>
      <Reference>
        <URL>https://bugzilla.suse.com/923176</URL>
        <Description>SUSE Bug 923176</Description>
      </Reference>
    </References>
  </Vulnerability>
</cvrfdoc>
